docs/en/04_Changelogs/4.0.0.md

```diff
@@ -92,6 +93,138 @@ For a full list of renamed classes, check the `.upgrade.yml` definitions in each
 
 The rename won't affect class-based permission codes or database table names.
 
+##### Use ::class whenever possible
+
+Prefer:
+ ```php
+ private static $has_one = [
+   'MyObject' => MyObject::class
+ ];
+ ```
+ to
+ 
+ ```php
+private static $has_one = [
+  'MyObject' => 'My\Project\MyObject'
+];
+```
+
+##### Use Injector wherever you can
+
+This can help with reducing changes from 3.x to 4.x by allowing your code to 
+continue to use `MyClassName`, while you tell the injector what the FQ class name is:
+
+```yaml
+SilverStripe\Core\Injector\Injector:
+  MyClassName:
+    class: Me\MyModule\MyClassName # fully qualified!
+```
+
+##### Namespaced table names
+
+Keep in mind that table names are namespaced too. To ease data migrations, use the `$table_name` config property of your `DataObject` subclasses.
```

Comment:
Again this isn't "upgrading to silverstripe 4" this is "how to namespace your code"